File: tests/fixtures/generic/vulnerable/taint_xpathi.cs
// @audit-fixture
// @rule: taint_xpathi
// @expected: detected
using System.Xml;
public class XmlController : Controller {
    public IActionResult Search() {
        var nodes = doc.SelectNodes("//user[@name='" + Request.Query["name"] + "']");
        return Ok(nodes?.Count);
    }
}

File: tests/fixtures/generic/vulnerable/taint_xpathi.java
// @audit-fixture
// @rule: taint_xpathi
// @expected: detected
import javax.xml.xpath.*;
public class XPathServlet extends HttpServlet {
    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws Exception {
        XPath xpath = XPathFactory.newInstance().newXPath();
        String result = xpath.evaluate("//user[@name='" + request.getParameter("name") + "']", doc);
    }
}
